In [30]:
import pandas as pd
df = pd.read_csv("C:/Users/cucur/Downloads/supermarket_sales.csv")

In [31]:
df.head(4)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   
3  123-19-1176      A     Yangon        Member    Male   

             Product line  Unit price  Quantity   Tax 5%     Total       Date  \
0       Health and beauty       74.69         7  26.1415  548.9715   1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200   3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255   3/3/2019   
3       Health and beauty       58.22         8  23.2880  489.0480  1/27/2019   

    Time      Payment    cogs  gross margin percentage  gross income  Rating  
0  13:08      Ewallet  522.83                 4.761905       26.1415     9.1  
1  10:29         Cash   76.40                 4.761905        3.8200     9.6  
2  13:23  Credit card  324.31                 4.761905       16.2155     7.4  
3  20:33      Ewallet  465.76                 4.761905       23.2880     8.4

In [32]:
df.tail(4)

Invoice ID Branch      City Customer type  Gender         Product line  \
996  303-96-2227      B  Mandalay        Normal  Female   Home and lifestyle   
997  727-02-1313      A    Yangon        Member    Male   Food and beverages   
998  347-56-2442      A    Yangon        Normal    Male   Home and lifestyle   
999  849-09-3807      A    Yangon        Member  Female  Fashion accessories   

     Unit price  Quantity  Tax 5%     Total       Date   Time  Payment  \
996       97.38        10  48.690  1022.490   3/2/2019  17:16  Ewallet   
997       31.84         1   1.592    33.432   2/9/2019  13:22     Cash   
998       65.82         1   3.291    69.111  2/22/2019  15:33     Cash   
999       88.34         7  30.919   649.299  2/18/2019  13:28     Cash   

       cogs  gross margin percentage  gross income  Rating  
996  973.80                 4.761905        48.690     4.4  
997   31.84                 4.761905         1.592     7.7  
998   65.82                 4.761905         3.291     4.1  
999  618.38                 4.761905        30.919     6.6

In [33]:
df[615:620]

Invoice ID Branch       City Customer type  Gender  \
615  509-29-3912      A     Yangon        Member  Female   
616  327-40-9673      B   Mandalay        Member    Male   
617  840-19-2096      C  Naypyitaw        Member    Male   
618  828-46-6863      A     Yangon        Member    Male   
619  641-96-3695      C  Naypyitaw        Member  Female   

               Product line  Unit price  Quantity   Tax 5%     Total  \
615       Sports and travel       38.72         9  17.4240  365.9040   
616       Sports and travel       72.60         6  21.7800  457.3800   
617  Electronic accessories       87.91         5  21.9775  461.5275   
618      Food and beverages       98.53         6  29.5590  620.7390   
619     Fashion accessories       43.46         6  13.0380  273.7980   

          Date   Time      Payment    cogs  gross margin percentage  \
615  3/20/2019  12:24      Ewallet  348.48                 4.761905   
616  1/13/2019  19:51         Cash  435.60                 4.761905   
617  3/14/2019  18:10      Ewallet  439.55                 4.761905   
618  1/23/2019  11:22  Credit card  591.18                 4.761905   
619   2/7/2019  17:55      Ewallet  260.76                 4.761905   

     gross income  Rating  
615       17.4240     4.2  
616       21.7800     6.9  
617       21.9775     4.4  
618       29.5590     4.0  
619       13.0380     8.5

In [34]:
df.dtypes

Invoice ID                  object
Branch                      object
City                        object
Customer type               object
Gender                      object
Product line                object
Unit price                 float64
Quantity                     int64
Tax 5%                     float64
Total                      float64
Date                        object
Time                        object
Payment                     object
cogs                       float64
gross margin percentage    float64
gross income               float64
Rating                     float64
dtype: object

In [40]:
df["gross income"].describe(percentiles = [0.01, 0.25, 0.50, 0.75, 0.99])

count    1000.000000
mean       15.379369
std        11.708825
min         0.508500
1%          0.913735
25%         5.924875
50%        12.088000
75%        22.445250
99%        45.250750
max        49.650000
Name: gross income, dtype: float64

75 персентиль нам показує,що 75% людей всередньому внесли суму 22.44525

In [45]:
prcent_of_invoices_in_cities = df.City.value_counts()
show = prcent_of_invoices_in_cities * 100 /1000
print(show)

City
Yangon       34.0
Mandalay     33.2
Naypyitaw    32.8
Name: count, dtype: float64


In [69]:
yangon_gross_income = df[df['City'] == 'Yangon']['gross income'].mean()
mandalay_gross_income = df[df['City'] == 'Mandalay']['gross income'].mean()

print(f"Середнє значення gross income у місті Yangon: {yangon_gross_income}")
print(f"Середнє значення gross income у місті Mandalay: {mandalay_gross_income}")

Середнє значення gross income у місті Yangon: 14.874001470588235
Середнє значення gross income у місті Mandalay: 15.232024096385542


Гіпотези:
1. Можливо населення міст є різним.
2. Можливо в цих містах є різна зацівленість в товарах які ми продаємо,тобто в одному з них менше цільової аудиторії ніж в іншому.
3. Можливо компанія має різні цінові політики в різних містах і тому в одному з них,товари дорожці, а спроможність покупця менша.

In [80]:
yangon_gross_income = df[df['City'] == 'Yangon']['gross margin percentage'].mean()
mandalay_gross_income = df[df['City'] == 'Mandalay']['gross margin percentage'].mean()
if yangon_gross_income == mandalay_gross_income:
    print("YES all gross margin percentage are equal")
else:
    print("NO all gross margin percentage are equal")

YES all gross margin percentage are equal


In [91]:
block = df[df["City"] == "Yangon"]["Product line"].value_counts()
print(block)

Product line
Home and lifestyle        65
Electronic accessories    60
Sports and travel         59
Food and beverages        58
Fashion accessories       51
Health and beauty         47
Name: count, dtype: int64


In [92]:
block = df[df["City"] == "Mandalay"]["Product line"].value_counts()
print(block)

Product line
Fashion accessories       62
Sports and travel         62
Electronic accessories    55
Health and beauty         53
Food and beverages        50
Home and lifestyle        50
Name: count, dtype: int64
